### DISCUSSION TITLE
Can't set background opacity with startup session

### DISCUSSION THREAD

**Discussion by 0xSigi** [kovidgoyal/kitty#4146] [Q&A] [answered]
Hi, not sure if this is indeed a bug, or such a feature was not implemented but I can't set background opacity when trying to set a startup session for kitty.. I need it for a 'quick' dropdown term used to for example take notes or work on top of other windows like browsers, so I need to see what's below. I tried all options listed in the documentation but it always returns same error

> ValueError: Unknown command in session file: 

How would I go about setting opacity in such a session file? While I can set it directly via sway, it blurs the entire window, including the boarders/bar so this is less than ideal solution here.

**Comment by page-down** [comment]
I suggest moving to GitHub Discussions.

You need the default launched window to be semi-transparent, but subsequent new os windows to be opaque, right?

You need to use the remote control feature because the session file is only a subset of it, for ease of use.
https://sw.kovidgoyal.net/kitty/remote-control/#kitty-set-background-opacity

The following configuration enables remote control with `socket-only`, which can only be operated locally with unix socket file access.

All initialization actions are no longer done in the session file, but with the remote control commands.

Please refer to the documentation for the meaning of the following configuration options.

kitty.conf
```text
startup_session startup_session.conf
allow_remote_control socket-only
listen_on unix:~/.local/kitty-{kitty_pid}.sock

dynamic_background_opacity yes
```

startup_session.conf
```text
launch /absolute/path/to/.config/kitty/startup.sh
```

startup.sh (chmod +x)
```
#!/usr/bin/env bash
kitty @ set-background-opacity 0.8
kitty @ launch --title do-whatever-you-want
```

**Comment by kovidgoyal** [comment, marked as answer]
Have your special transparent instance run like this

```
kitty -o background_opacity=0.5 --session /path/to/session.conf
```

**Reply by 0xSigi** [reply]
Thanks a lot! This indeed works as expected, unfortunately solution provided by @page-down had a weird issue that the kitty window would disappear after few seconds.. Haven't found anything in logs, and did not had the time to deep further into it as this works perfectly.
